src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -132,11 +132,19 @@ impl<Src: OpendalSource> OpendalReader<Src> {
         let source_name = source_ctx.source_name.clone();
         let object_name = split.name.clone();
         let start_offset = split.offset;
-        let reader = op
-            .read_with(&object_name)
-            .range(start_offset as u64..)
-            .into_future() // Unlike `rustc`, `try_stream` seems require manual `into_future`.
-            .await?;
+        // After a recovery occurs, for gzip-compressed files, it is necessary to read from the beginning each time,
+        // other files can continue reading from the last read `start_offset`.
+        let reader = match object_name.ends_with(".gz") || object_name.ends_with(".gzip") {
```

Comment:
In addition to suffix check, should we also check `compression_format` (See L152)?

Reply:
I think no need, but we can add a check: if compression_format is set, suffix must be `.gz` or `.gzip`.